shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -917,24 +899,25 @@ - (void)viewWillTransitionToSize:(CGSize)size
     return;
   }
 
+  __weak FlutterViewController* weakSelf = self;
   _shouldIgnoreViewportMetricsUpdatesDuringRotation = YES;
   dispatch_after(dispatch_time(DISPATCH_TIME_NOW,
                                static_cast<int64_t>(transitionDuration / 2.0 * NSEC_PER_SEC)),
                  dispatch_get_main_queue(), ^{
                    // `viewWillTransitionToSize` is only called after the previous rotation is
                    // complete. So there won't be race condition for this flag.
                    _shouldIgnoreViewportMetricsUpdatesDuringRotation = NO;
```

Comment:
weakSelf->_shouldIgnoreViewportMetricsUpdatesDuringRotation ?

Or is this retaining `self` on purpose?  @hellohuanlin wrote this #40730

Reply:
Oh yikes! Totally unintentional! Nice catch.